raiden/tests/unit/rif_comms/test_rif_comms_client.py

```diff
@@ -53,70 +60,57 @@ class TestRiffCommsClient(unittest.TestCase):
-        peer_id = self.rif_comms_client._get_peer_id(LUMINO_1_ADDRESS)
-        channel = self.rif_comms_client.subscribe_to(get_random_address_str())
-        peer_id = self.rif_comms_client._get_peer_id(LUMINO_1_ADDRESS)
+        notification = self.rif_comms_client1.connect()
+        channel = self.rif_comms_client1.subscribe_to(get_random_address_str())
+        peer_id = self.rif_comms_client1._get_peer_id(LUMINO_3_ADDRESS)
 
-    @pytest.mark.skip(reason="ignore")
-    def test_subscribe(self):
-        # TODO: this is failing with "peer is not subscribed to"
-        channel = grpc.insecure_channel(LUMINO_1_COMMS_API)
-        stub = CommunicationsApiStub(channel)
-        rsk_address = RskAddress(address=LUMINO_1_ADDRESS)
-        notification = stub.ConnectToCommunicationsNode(rsk_address)
-        channel = stub.CreateTopicWithRskAddress(rsk_address)
-        subscribers = stub.GetSubscribers(Channel(channelId=LUMINO_1_ADDRESS))
 
-    @pytest.mark.skip(reason="ignore")
+    @pytest.mark.skip(reason="rif comms issue infinite loop")
     def test_has_subscriber(self):
-        # TODO: test with RSK address instead of peer ID
-        channel = grpc.insecure_channel(LUMINO_1_COMMS_API)
-        stub = CommunicationsApiStub(channel)
-        rsk_address = RskAddress(address=LUMINO_1_ADDRESS)
-        notification = stub.ConnectToCommunicationsNode(rsk_address)
-        channel = stub.CreateTopicWithPeerId(rsk_address)
-        peer_id = stub.LocatePeerId(rsk_address)
-
-        """
-        message Subscriber {
-            string peerId = 1;
-            Channel channel = 2;
-        }
-        """
-        has_subscriber = stub.HasSubscriber(
-            Subscriber(
-                peerId=peer_id.address,
-                channel=Channel(channelId=LUMINO_1_ADDRESS)
-            )
-        )
-        print("has_subscriber", has_subscriber)
+        notification = self.rif_comms_client1.connect()
+        notification2 = self.rif_comms_client2.connect()
+        channel = self.rif_comms_client1.subscribe_to(LUMINO_2_ADDRESS)
+        time.sleep(10)
+        subscribed = self.rif_comms_client1.is_subscribed_to(LUMINO_2_ADDRESS)
+        print("Subscribed: ", subscribed)
+
+
+
+    @pytest.mark.skip(reason="works but subscribed equals False")
+    def test_has_subscriber_self(self):
+        notification = self.rif_comms_client1.connect()
+        channel = self.rif_comms_client1.subscribe_to(LUMINO_1_ADDRESS)
+        subscribed = self.rif_comms_client1.is_subscribed_to(LUMINO_1_ADDRESS)
+        print("Subscribed: ", subscribed)
 
     @pytest.mark.skip(reason="ignore")
     def test_disconnect(self):
         notification = self.rif_comms_client.connect()
         peer_id = self.rif_comms_client._get_peer_id(LUMINO_1_ADDRESS)
         self.rif_comms_client.disconnect()
+        # TODO check if end comms deletes topics
 
     @pytest.mark.skip(reason="ignore")
     def test_send_lumino_message(self):
```

Comment:
would you agree on not having a test which requires an entire lumino node to be fired up?

if so, i'll create an issue to mock or somehow reduce this

Reply:
the purpose was to test that a running node of lumino can receive and parse correctly a message sent to a rif comms node

i agree we need to mock this or create an integration test that really runs a lumino node, but for now i fund this test useful

Reply:
https://jirainfuy.atlassian.net/browse/RLP-910